imap-proto/src/types.rs

```diff
@@ -118,13 +118,97 @@ pub enum SectionPath {
     Part(Vec<u32>, Option<MessageSection>),
 }
 
+pub type BodyParam<'a> = Option<Vec<(&'a str, &'a str)>>;
```

Comment:
I'd like the order here to be reversed: `BodyParam` at the bottom (no dependencies, used by the other types), and `BodyStructure` should come below `AttributeValue`. (See also https://deterministic.space/how-to-order-rust-code.html.)